src/xregexp.js

```diff
@@ -51,7 +51,10 @@ var nativeTokens = {
     'class': /\\(?:[0-3][0-7]{0,2}|[4-7][0-7]?|x[\dA-Fa-f]{2}|u(?:[\dA-Fa-f]{4}|{[\dA-Fa-f]+})|c[A-Za-z]|[\s\S])|[\s\S]/
 };
 // Any backreference or dollar-prefixed character in replacement strings
-var replacementToken = /\$(?:{([\w$]+)}|(\d\d?|[\s\S]))/g;
+var replacementTokens = [
+    /\$(?:<([\w$]+)>)/g,
+    /\$(?:{([\w$]+)}|(\d\d?|[\s\S]))/g
+];
```

Comment:
Ah, that's exactly right! Thanks for the suggestion, I've pushed some updates implementing it.